Octokit/Models/Request/SearchCommitsRequest.cs

```diff
@@ -0,0 +1,291 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using Octokit.Internal;
+
+namespace Octokit
+{
+    /// <summary>
+    /// Search commits
+    /// </summary>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class SearchCommitsRequest : BaseSearchRequest
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchCommitsRequest"/> class.
+        /// </summary>
+        public SearchCommitsRequest() : base()
+        {
+            Repos = new RepositoryCollection();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchCommitsRequest"/> class.
+        /// </summary>
+        /// <param name="term">The search term.</param>
+        public SearchCommitsRequest(string term) : base(term)
+        {
+            Repos = new RepositoryCollection();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchCommitsRequest"/> class.
+        /// </summary>
+        /// <param name="term">The term.</param>
+        /// <param name="owner">The owner.</param>
+        /// <param name="name">The name.</param>
+        public SearchCommitsRequest(string term, string owner, string name)
+            : this(term)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, nameof(owner));
+            Ensure.ArgumentNotNullOrEmptyString(name, nameof(name));
+
+            Repos.Add(owner, name);
+        }
+
+        /// <summary>
+        /// Optional Sort field. Can only be by author-date or committer-date. 
+        /// If not provided, results are sorted by best match.
+        /// </summary>
+        /// <remarks>
+        /// http://developer.github.com/v3/search/#search-commits
+        /// </remarks>
+        public CommitsSearchSort? SortField { get; set; }
+        public override string Sort
+        {
+            get { return SortField.ToParameter(); }
+        }
+
+        private string QueryDateTime(DateTime input)
+        {
+            return input.ToString("yyyy-MM-DD");
```

Comment:
🤔 I need to have another look at places where we need to handle date inputs, and see if this is a chance for us to standardize on date input.

Reply:
@shiftkey have you decided on this?

Reply:
@shiftkey Any update?